sklearn/linear_model/base.py

```diff
@@ -56,12 +57,19 @@ def make_dataset(X, y, sample_weight, random_state=None):
     # seed should never be 0 in SequentialDataset
     seed = rng.randint(1, np.iinfo(np.int32).max)
 
+    if X.dtype == np.float32:
+        CSRData = CSRDataset32
+        ArrayData = ArrayDataset32
```

Comment:
The [codecov chrome/firefox extension](https://github.com/codecov/browser-extension) tells me that those lines are not covered. Please add a test for that case (and install the codecov browser extension ;).

This test should probably fit a `SGDClassifier` on 32 bit float iris and check that the `coef_` attribute should be 32 bit float as well ( and the output of `decision_function` should also output a float32 array).